File: SretneSapice/SretneSapice.Services/Database/DogWalkerAvailability.cs
using System;
using System.Collections.Generic;

namespace SretneSapice.Services.Database;

public partial class DogWalkerAvailability
{
    public int DogWalkerId { get; set; }

    public DateTime Date { get; set; }

    public DateTime StartTime { get; set; }

    public DateTime EndTime { get; set; }

    public string? AvailabilityStatus { get; set; }

    public virtual DogWalker DogWalker { get; set; } = null!;
}

File: SretneSapice/UI/sretnesapice_mobile/lib/models/dog_walker_availability.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sretnesapice_mobile/models/dog_walker.dart';

part 'dog_walker_availability.g.dart';

@JsonSerializable()
class DogWalkerAvailability {
  int? dogWalkerId;
  DateTime? date;
  DateTime? startTime;
  DateTime? endTime;
  String? availabilityStatus;
  DogWalker? dogWalker;

  DogWalkerAvailability(
    this.dogWalkerId,
    this.date,
    this.startTime,
    this.endTime,
    this.availabilityStatus,
    this.dogWalker,
  );

  factory DogWalkerAvailability.fromJson(Map<String, dynamic> json) =>
      _$DogWalkerAvailabilityFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$DogWalkerAvailabilityToJson(this);
}
